metrics.go

```diff
@@ -78,140 +69,110 @@ type PrometheusMetrics struct {
 	registry           prometheus.Registerer
 	totalThreads       prometheus.Counter
 	busyThreads        prometheus.Gauge
-	totalWorkers       map[string]prometheus.Gauge
-	busyWorkers        map[string]prometheus.Gauge
-	readyWorkers       map[string]prometheus.Gauge
-	workerCrashes      map[string]prometheus.Counter
-	workerRestarts     map[string]prometheus.Counter
-	workerRequestTime  map[string]prometheus.Counter
-	workerRequestCount map[string]prometheus.Counter
+	totalWorkers       *prometheus.GaugeVec
+	busyWorkers        *prometheus.GaugeVec
+	readyWorkers       *prometheus.GaugeVec
+	workerCrashes      *prometheus.CounterVec
+	workerRestarts     *prometheus.CounterVec
+	workerRequestTime  *prometheus.CounterVec
+	workerRequestCount *prometheus.CounterVec
```

Comment:
At least until 2.0, we will need to probably support both ways for BC. Otherwise, anyone using the current metrics will have no metrics on the next release.

Ultimately, it's @dunglas's call though.

Reply:
v1.4.0 was released last month, and no metrics, i didn't see anyone raise an issue (it is maybe user who utilizes metrics is in a small percentage or have not upgraded to v1.4) hence i think modifying the metrics quite safe, i know in software development it is BC break

Reply:
That is a fair assessment. :)

We should probably update the documentation in this PR too.